certbot-apache/certbot_apache/configurator.py

```diff
@@ -959,9 +968,100 @@ def _copy_create_ssl_vhost_skeleton(self, avail_fp, ssl_fp):
+                    vhost.filep, ssl_fp), reporter.MEDIUM_PRIORITY)
+        self.aug.set("/augeas/files%s/mtime" % (self._escape(ssl_fp)), "0")
+        self.aug.set("/augeas/files%s/mtime" % (self._escape(vhost.filep)), "0")
+
+    def _prepare_ssl_skeleton_contents(self, contents):
+        """ Helper function for _copy_create_ssl_vhost_skeleton to prepare the
+        new HTTPS VirtualHost contents. Currently disabling the rewrites """
+
+        result = []
+        sift = False
+        contents = iter(contents)
+
+        comment = ("# Some rewrite rules in this file were "
+                   "disabled on your HTTPS site,\n"
+                   "# because they have the potential to create "
+                   "redirection loops.\n")
+
+        for line in contents:
+            A = line.lower().lstrip().startswith("rewritecond")
+            B = line.lower().lstrip().startswith("rewriterule")
+
+            if not (A or B):
+                result.append(line)
+                continue
+
+            # A RewriteRule that doesn't need filtering
+            if B and not self._sift_rewrite_rule(line):
+                result.append(line)
+                continue
+
+            # A RewriteRule that does need filtering
+            if B and self._sift_rewrite_rule(line):
+                if not sift:
+                    result.append(comment)
+                    sift = True
+                result.append("# " + line)
+                continue
+
+            # We save RewriteCond(s) and their corresponding
+            # RewriteRule in 'chunk'.
+            # We then decide whether we comment out the entire
+            # chunk based on its RewriteRule.
+            chunk = []
+            if A:
+                chunk.append(line)
+                line = next(contents)
+
+                # RewriteCond(s) must be followed by one RewriteRule
+                while not line.lower().lstrip().startswith("rewriterule"):
+                    chunk.append(line)
+                    line = next(contents)
+
+                # Now, current line must start with a RewriteRule
+                chunk.append(line)
+
+                if self._sift_rewrite_rule(line):
+                    if not sift:
+                        result.append(comment)
+                        sift = True
+
+                    result.append('\n'.join(
+                        ['# ' + l for l in chunk]))
+                    continue
+                else:
+                    result.append('\n'.join(chunk))
+                    continue
+        return result, sift
+
+    def _get_vhost_block(self, vhost):
+        """ Helper method to get VirtualHost contents from the original file.
+        This is done with help of augeas span, which returns the span start and
+        end positions
+
+        :returns: `list` of VirtualHost block content lines without closing tag
+        """
+
+        try:
+            span_val = self.aug.span(vhost.path)
```

Comment:
I think there may be a problem here. This copies the vhost and ignores any other directives in the file. For example, a user may have:
```
<IfModule mod_macro.c>
    <VirtualHost *:80>
        ....
    </VirtualHost>
</IfModule>
```
When we copy the vhost, we remove these `<IfModule>` lines which seems incorrect. To be clear, this is the case for all directives that start a block not just `<IfModule>`.

One thing I'm not sure about is if directives contained in the block that contains the vhost should be copied. To provide a toy example because to be honest I'm not familiar enough with Apache directives to be aware of a real one:
```
<IfModule mod_magic.c>
    MagicSpell "foo"
    <VirtualHost *:80>
        ....
    </VirtualHost>
</IfModule>
```
Does `MagicSpell "foo"` apply to the Apache config globally or that specific block? If the former, we ideally shouldn't copy them but we probably should if it's the latter. Is it dependent on the directive?

I'm not sure how best to handle this. Perhaps we could copy everything in the highest level block that contains the vhost which shouldn't be too hard using our functions to manipulate Augeas paths and `span` but the two problems I see are:

1. Directives that apply globally we'd be duplicating. We've been doing this for years though by copying the entire file containing the vhost so maybe this isn't a problem?
2. There may be other vhosts in that block which we don't want to copy. We'd have to remove these somehow which will probably be a bit tricky.

I'm curious your thoughts on this.